### PR TITLE
Mqtt add object_id

### PR DESCRIPTION
**Pull request by Smeagolworms4**
## Proposed change

Add documentation of object_id in relation with with PR: https://github.com/home-assistant/core/pull/58728

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/58728
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/mqtt/discovery.markdown [hunk 1]
```diff
@@ -481,3 +482,22 @@ If the device supports gps coordinates then they can be sent to Home Assistant b
   "gps_accuracy": 1.2
  }
 ```
+
+### Used object_id to generate the entity id
```
Comment (emontnemery) [inline source/_docs/mqtt/discovery.markdown]:
```suggestion
### Use object_id to influence the entity id
```

File: source/_integrations/light.mqtt.markdown [hunk 2]
```diff
@@ -230,6 +230,10 @@ name:
   required: false
   type: string
   default: MQTT Light
+object_id:
```
Comment (emontnemery) [inline source/_integrations/light.mqtt.markdown:233]: Please update all three schemas (basic, json, template) with the new parameter.

File: source/_integrations/vacuum.mqtt.markdown [hunk 3]
```diff
@@ -140,6 +140,10 @@ name:
   required: false
   type: string
   default: MQTT Vacuum
+object_id:
```
Comment (emontnemery) [inline source/_integrations/vacuum.mqtt.markdown:143]: Please update both schemas (legacy, state) with the new parameter.

Comment (Smeagolworms4) [reply source/_integrations/vacuum.mqtt.markdown:143]: Done (add in state)

**Review by emontnemery — APPROVED** [review]
Thanks, @Smeagolworms4 👍

**Comment by frenck** [comment]
This PR has been targeting and merged into the wrong branch.

This is a new feature, thus should go into the `next` branch. As the bot pointed out here:
<https://github.com/home-assistant/home-assistant.io/pull/20217#issuecomment-963118589>

As a result, this property is now visible in the current documentation.

**Comment by emontnemery** [comment]
Oops, sorry about that @frenck.

Reverted by https://github.com/home-assistant/home-assistant.io/pull/20233

**Comment by frenck** [comment]
Should we now revert the revert on `next`? :)

**Comment by emontnemery** [comment]
Yes, new PR here: https://github.com/home-assistant/home-assistant.io/pull/20234
Edit: That didn't work, the branch includes an unwanted commit

**Comment by emontnemery** [comment]
New attempt: https://github.com/home-assistant/home-assistant.io/pull/20235

**Pull request merged**